File: src/main/java/at/searles/regexp/Union.java
package at.searles.regexp;

class Union implements Regexp {

    private final Regexp l;
    private final Regexp r;

    Union(Regexp l, Regexp r) {
        this.l = l;
        this.r = r;
    }

    public String toString() {
        return "or(" + l.toString() + ", " + r.toString() + ")";
    }

    @Override
    public <A> A accept(Visitor<A> visitor) {
        return visitor.visitOr(l, r);
    }
}

File: src/main/kotlin/at/searles/parsing/lexer/regexp/Union.kt
package at.searles.parsing.lexer.regexp

internal class Union(private val l: Regexp, private val r: Regexp) : Regexp {
    override fun toString(): String {
        return "$l | $r"
    }

    override fun <A> accept(visitor: Visitor<A>): A {
        return visitor.visitOr(l, r)
    }

}